File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        
        ArrayList<Integer> result = new ArrayList<>();
        int p1 = 0, p2 = 0;
        
        
        while (p1 < m && p2 < n) {
            if (nums1[p1] <= nums2[p2]) {
                result.add(nums1[p1]);
                p1 += 1;
            } else if(nums1[p1] > nums2[p2]) {
                result.add(nums2[p2]);
                p2 += 1;
            }
        }
        
        // 왼쪽이 남은 경우
        if (p2 == n) {
            while (p1 < m) {
                result.add(nums1[p1]);
                p1 += 1;
            }
            
        // 오른쪽이 남은 경우
        } else if (p1 == m) {
            while (p2 < n) {
                result.add(nums2[p2]);
                p2 += 1;
            }
        }
        
        
        for (int i = 0; i < m + n; i++) {
            nums1[i] = result.get(i);
        }
        
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:

        # 병합정렬 -> nums1에 넣어주기
        # [1, 2, 3, 0, 0, 0] [2, 5, 6]

        p1, p2, idx = m - 1, n - 1, m + n - 1
        
        while p1 >= 0  and p2 >= 0:
            if nums1[p1] >= nums2[p2]:
                nums1[idx] = nums1[p1]
                idx -= 1
                p1 -= 1
            elif nums1[p1] < nums2[p2]:
                nums1[idx] = nums2[p2]
                idx -= 1
                p2 -= 1
        
        # 왼쪽이 남는경우
        if p2 < 0:
            while p1 >= 0:
                nums1[idx] = nums1[p1]
                idx -= 1
                p1 -= 1
        # 오른쪽이 남는경우
        elif p1 < 0:
            while p2 >= 0:
                nums1[idx] = nums2[p2]
                idx -= 1
                p2 -= 1